### PR TITLE
CD-236755 - master - added a presence check for data

### PR DESCRIPTION
**Pull request by rohitcoupa**
In :results_option_build , application is throwing error not type checking.
fixed the type check before using the data.

### CODE REVIEW

**Comment by rohitcoupa** [comment]
@johnny-lai Please review

File: coffee/lib/abstract-chosen.coffee [hunk 1]
```diff
@@ -95,7 +95,7 @@ class AbstractChosen
       for data in this.get_selected_items()
         if @is_multiple
           this.choice_build data
-        else
+        else if data
```
Comment (johnny-lai) [inline coffee/lib/abstract-chosen.coffee:98]: If `data` is empty, it probably  shouldn't be in the `this.get_selected_items()` in the first place. Why is `data` empty and in the list of selected items?

What is the error that is encountered?

Does this error also happen in the `@is_multiple` case?

**Comment by johnny-lai** [comment]
To build the JavaScript, you use `grunt`.

See https://github.com/coupa/chosen/blob/master/Gruntfile.coffee for more information.

File: coffee/lib/abstract-chosen.coffee (hunk 1 above)
Comment (rohitcoupa) [reply coffee/lib/abstract-chosen.coffee:98]: >Does this error also happen in the @is_multiple case?

No,
it comes into the else condition.

`_ref3 = this.get_selected_items();`
=> [undefined]

On Easy Forms, when I enter data for Commodities and select one.
Screenshot 2021-03-08 at 10 34 17 PM


It adds these Options into HTML
Screenshot 2021-03-08 at 10 33 21 PM

And When I click on "ADD" button I get thrown the following error for
data.text
110122625-dafe1700-7de5-11eb-9d4f-2ddd3568eced

Comment (rohitcoupa) [reply coffee/lib/abstract-chosen.coffee:98]: The PR I had raised fixed the Issue.
https://github.com/coupa/coupa_development/pull/82930/files

**Comment by rohitcoupa** [comment]
I have built the JS, and gave a PR to the enterprise.
https://github.com/coupa/coupa_development/pull/82930

File: coffee/lib/abstract-chosen.coffee (hunk 1 above)
Comment (johnny-lai) [reply coffee/lib/abstract-chosen.coffee:98]: So `data` is `undefined`. Why is it in the list of selected_items?

Comment (johnny-lai) [reply coffee/lib/abstract-chosen.coffee:98]: It doesn't make sense for an "undefined" item to be selected to me.

Comment (rohitcoupa) [reply coffee/lib/abstract-chosen.coffee:98]: I have put loggers on this clone instance.
https://clone-macquarie-sf-00744675rajjo.coupadev.com/easy_form_responses/start?easy_form_id=459

```
_ref3 = this.get_selected_items();
        console.log('--------------- _ref3 is:', _ref3);
for (_j = 0, _len1 = _ref3.length; _j < _len1; _j++) {
    data = _ref3[_j];
    console.log('--------------- data is:', undefined);
}
```

This is the outcome.
Screenshot 2021-03-08 at 11 10 00 PM

Comment (johnny-lai) [reply coffee/lib/abstract-chosen.coffee:98]: I understand that there is `undefined` in the selected list. I'm asking you to figure out why so we can fix the root cause and not the symptom

File: coffee/chosen.jquery.coffee [hunk 2]
```diff
@@ -644,7 +644,12 @@ class Chosen extends AbstractChosen
         [ val ]
 
     if val?
-      (@source.get_item_by_value(v) for v in val)
+      for v in val
+        item_by_val = @source.get_item_by_value(v)
+        if (item_by_val != undefined)
+          item_by_val
+        else
+
```
Comment (johnny-lai) [inline coffee/chosen.jquery.coffee:652]: Looks a little weird.. but I did try locally and it seems to work
```
~/Coupa/Scratch
$ cat test.coffee 
x = () ->
  val = [1, undefined, 3]

  for v in val
    item_by_val = v
    if item_by_val != undefined
      item_by_val
    else

console.log x()
$ coffee test.coffee 
[ 1, 3 ]
```

**Comment by johnny-lai** [comment]
lgtm

**Comment by johnny-lai** [comment]
[lockit v4.9.3] merged via Lockit CLI on 2021-03-11 08:53:37.472787 -0800 PST m=+5.559681023

**Pull request merged**